src/sql/workbench/parts/notebook/browser/models/notebookModel.ts

```diff
@@ -1007,6 +1012,8 @@ export class NotebookModel extends Disposable implements INotebookModel {
 		switch (change) {
 			case NotebookChangeType.CellOutputUpdated:
 			case NotebookChangeType.CellSourceUpdated:
+			case NotebookChangeType.CellInputVisibilityChanged:
+				changeInfo.changeType = NotebookChangeType.DirtyStateChanged;
```

Comment:
does this need to be added for all of these change types?

Reply:
Looking into it.